streaming/base/storage/upload.py

```diff
@@ -261,19 +274,34 @@ def __init__(self,
                  progress_bar: bool = False) -> None:
         super().__init__(out, keep_local, progress_bar)
 
-        import boto3
+        if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
+            from google.cloud.storage import Client
+
+            service_account_path = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+            self.gcs_client = Client.from_service_account_json(service_account_path)
+            self.authentication = Authentication.SERVICE_ACCOUNT
+        elif 'GCS_KEY' in os.environ and 'GCS_SECRET' in os.environ:
+            import boto3
+
+            # Create a session and use it to make our client. Unlike Resources and Sessions,
+            # clients are generally thread-safe.
+            session = boto3.session.Session()
+            self.gcs_client = session.client(
+                's3',
+                region_name='auto',
+                endpoint_url='https://storage.googleapis.com',
+                aws_access_key_id=os.environ['GCS_KEY'],
+                aws_secret_access_key=os.environ['GCS_SECRET'],
+            )
+            self.authentication = Authentication.HMAC
+        else:
+            raise ValueError(('Either GOOGLE_APPLICATION_CREDENTIALS needs to be set for'
```

Comment:
Extra `(` and `)` ?

Reply:
If I don't have this pylance complains about implicit string concatenation.

Reply:
Changed it to be a single string so extra parentheses aren't needed, but I think the previous version looked more readable.

Reply:
How about putting [this](https://github.com/mosaicml/streaming/blob/main/streaming/base/stream.py#L191) way ?

Reply:
Done.